modules/apps/frontend-data-set/frontend-data-set-admin-web/src/main/java/com/liferay/frontend/data/set/admin/web/internal/portlet/action/GetSystemDataSetsMVCResourceCommand.java

```diff
@@ -0,0 +1,150 @@
+import com.liferay.object.model.ObjectEntry;
+import com.liferay.object.service.ObjectDefinitionLocalService;
+import com.liferay.object.service.ObjectEntryLocalService;
+import com.liferay.portal.kernel.json.JSONFactory;
+import com.liferay.portal.kernel.json.JSONUtil;
+import com.liferay.portal.kernel.portlet.JSONPortletResponseUtil;
+import com.liferay.portal.kernel.portlet.bridges.mvc.BaseMVCResourceCommand;
+import com.liferay.portal.kernel.portlet.bridges.mvc.MVCResourceCommand;
+import com.liferay.portal.kernel.theme.ThemeDisplay;
+import com.liferay.portal.kernel.util.ParamUtil;
+import com.liferay.portal.kernel.util.Portal;
+import com.liferay.portal.kernel.util.StringUtil;
+import com.liferay.portal.kernel.util.WebKeys;
+
+import java.util.Set;
+
+import javax.portlet.ResourceRequest;
+import javax.portlet.ResourceResponse;
+
+import javax.servlet.http.HttpServletRequest;
+
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+
+/**
+ * @author Kevin Tan
+ */
+@Component(
+	property = {
+		"javax.portlet.name=" + FDSAdminPortletKeys.FDS_ADMIN,
+		"mvc.command.name=/frontend_data_set_admin/get_system_data_sets"
+	},
+	service = MVCResourceCommand.class
+)
+public class GetSystemDataSetsMVCResourceCommand
+	extends BaseMVCResourceCommand {
+
+	@Override
+	protected void doServeResource(
+			ResourceRequest resourceRequest, ResourceResponse resourceResponse)
+		throws Exception {
+
+		Set<String> systemFDSNames =
+			_systemFDSEntryRegistry.getSystemFDSNames();
+
+		if (systemFDSNames == null) {
+			JSONPortletResponseUtil.writeJSON(
+				resourceRequest, resourceResponse,
+				JSONUtil.put("items", _jsonFactory.createJSONArray()));
+
+			return;
+		}
+
+		ThemeDisplay themeDisplay = (ThemeDisplay)resourceRequest.getAttribute(
+			WebKeys.THEME_DISPLAY);
+
+		ObjectDefinition dataSetObjectDefinition =
+			_objectDefinitionLocalService.
+				getObjectDefinitionByExternalReferenceCode(
+					"L_DATA_SET", themeDisplay.getCompanyId());
+
+		HttpServletRequest httpServletRequest =
+			_portal.getOriginalServletRequest(
+				_portal.getHttpServletRequest(resourceRequest));
+
+		String search = ParamUtil.getString(httpServletRequest, "search");
+
+		JSONPortletResponseUtil.writeJSON(
+			resourceRequest, resourceResponse,
+			JSONUtil.put(
+				"items",
+				JSONUtil.toJSONArray(
+					systemFDSNames,
+					systemFDSName -> {
+						SystemFDSEntry systemFDSEntry =
+							_systemFDSEntryRegistry.getSystemFDSEntry(
+								systemFDSName);
+
+						if (!StringUtil.matchesIgnoreCase(
+								systemFDSEntry.getTitle(), search)) {
+
+							return null;
+						}
+
+						ObjectEntry objectEntry =
+							_objectEntryLocalService.fetchObjectEntry(
+								systemFDSEntry.getName(),
+								dataSetObjectDefinition.
+									getObjectDefinitionId());
+
+						boolean created = false;
+
+						if (objectEntry != null) {
+							created = true;
+						}
```

Comment:
This can be simplified. You can just use `objectEntry != null` inline, no need for a variable.

Would it be more accurate to call the prop `imported`?

Reply:
I'll move this inline. I just thought this was preferred since I noticed this pattern in java more often and I don't think I've seen this inlined, but I'll give it a go since it does seem cleaner.

Reply:
Having this property called `imported` makes sense to me though since the MVCResourceCommand calls this action "import" so I'll make this change as well